### PR TITLE
Make non-vararg dialect ops getter const

### PR DESCRIPTION
**Pull request by tsymalla**

### CODE REVIEW

File: lib/TableGen/Operations.cpp [hunk 1]
```diff
@@ -202,6 +209,13 @@ void AccessorBuilder::emitGetterDefinition() const {
         return $fromLlvm;
       })",
                 &m_fmt);
+  if (!m_arg.type->isVarArgList()) {
+    m_os << tgfmt(R"(
+      $cppType $_op::get$Name() const {
+        return $fromLlvm;
+      })",
+                  &m_fmt);
+  }
```
[the remark this answers is not in the document]
Comment (tsymalla-AMD) [reply lib/TableGen/Operations.cpp:218]: How are we going to chain calls like getSomething()->setSomeVal()` w/o a non-const getter?

[the remark this answers is not in the document]
Comment (Flakebi) [reply lib/TableGen/Operations.cpp:218]: Shouldn’t that just work with a const getter? The return type is non-const anyway.

[the remark this answers is not in the document]
Comment (tsymalla-AMD) [reply lib/TableGen/Operations.cpp:218]: For me, it doesn't, see the `static_assert` that tries to invoke a setter

[the remark this answers is not in the document]
Comment (Flakebi) [reply lib/TableGen/Operations.cpp:218]: Seems to work for me 😉

https://github.com/GPUOpen-Drivers/llvm-dialects/commit/d26ac6751a9be134838e5a77f729e836cf1c6eed

[the remark this answers is not in the document]
Comment (tsymalla-AMD) [reply lib/TableGen/Operations.cpp:218]: Then just let's make the getters const

File: example/ExampleMain.cpp [hunk 2]
```diff
@@ -91,7 +91,7 @@ void createFunctionExample(Module &module, const Twine &name) {
   Value *x2 = b.create<xd::Add32Op>(x1, sizeOf32, 11);
   Value *x3 = b.create<xd::CombineOp>(x2, x1);
   Value *x4 = b.create<xd::IExtOp>(b.getInt64Ty(), x3);
-  b.create<xd::WriteOp>(x4);
+  [[maybe_unused]] const xd::WriteOp *x5 = b.create<xd::WriteOp>(x4);
```
Comment (Flakebi) [inline example/ExampleMain.cpp:94]: The change here does not do anything?

**Review by Flakebi — COMMENTED** [review]
LGTM with a nit

File: example/ExampleMain.cpp (hunk 2 above)
Comment (tsymalla) [reply example/ExampleMain.cpp:94]: I forgot to change it. Will revert that change later and then merge the PR.

**Review by tsymalla-AMD — APPROVED** [review]

**Pull request merged**